### PR TITLE
add sub module export

### PR DESCRIPTION
**Pull request by doxiaodong**
* add sub module export
* see this issue https://github.com/DefinitelyTyped/DefinitelyTyped/issues/13337
* egg
```
import md5 = require('crypto-js/md5')
import sha1 = require('crypto-js/sha1')
...
```
* the export list is dependent on https://github.com/brix/crypto-js/tree/3.1.4#usage-without-requirejs

Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by misak113** [comment]
👍

**Comment by doxiaodong** [comment]
@misak113 , is it possible to merge with https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/cryptojs/index.d.ts? I don't think it's necessary to have two types of `crypto-js`

**Comment by misak113** [comment]
Probably yes however it is not under my privileges. Correct name based on npm registry is this crypto-js.

File: crypto-js/index.d.ts [hunk 1]
```diff
@@ -3,9 +3,6 @@
 // Definitions by: Michael Zabka <https://github.com/misak113/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-export = CryptoJS;
-export as namespace CryptoJS;
```
Comment (DanielRosenwasser) [inline crypto-js/index.d.ts]: This isn't the appropriate fix. These should be restored.

Then, each of the below modules that you've declared should have their own declaration file in the directory that import the necessary types from `index.d.ts` and export whatever they usually do.

So, for instance, you should have a `core.d.ts` in the `crypto-js` directory.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: crypto-js/index.d.ts (hunk 1 above)
Comment (doxiaodong) [inline crypto-js/index.d.ts]: I create `aes.d.ts`, and it works fine
```
import { AES } from './index'
export = AES;

```
But when I create `enc-base64.d.ts`, it throw `Cannot find namespace 'enc'`, how to fix it?
```
import { enc } from './index'
export = enc.Base64;
```

**Comment by doxiaodong** [comment]
@DanielRosenwasser , hello, thanks for review.

**Comment by DanielRosenwasser** [comment]
Thanks so much for putting the time in for this! I know that I've asked for a lot at this moment, but it'd be greatly appreciated if you had some basic tests for a few of these (i.e. just importing a few, utilizing the functionality, etc.)

**Comment by doxiaodong** [comment]
@DanielRosenwasser , I just copy `crypto-js-tests.ts` to `test/submodule-tests.ts`,  and use submodule api.

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**